app/src/androidTest/java/com/duckduckgo/app/browser/downloader/UriUtilsFilenameExtractorInternalTest.kt

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright (c) 2020 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.downloader
+
+import androidx.test.filters.SdkSuppress
+import com.duckduckgo.app.pixels.AppPixelName
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.appbuildconfig.api.AppBuildConfig
+import com.duckduckgo.appbuildconfig.api.BuildFlavor
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.mockito.kotlin.whenever
+
+class UriUtilsFilenameExtractorInternalTest {
```

Comment:
Existing tests for the original approach for extracting the file name. It is currently duplicated to test the new approach (see the internal flavor). Once the code will move from internal this file will be removed as we only need one set of tests.